flash/image/style_transfer/model.py

```diff
@@ -0,0 +1,174 @@
+from typing import Any, cast, Dict, List, Mapping, NoReturn, Optional, Sequence, Tuple, Type, Union
+
+import torch
+from torch import nn
+from torch.optim.lr_scheduler import _LRScheduler
+
+from flash.core.data.data_source import DefaultDataKeys
+from flash.core.data.process import Serializer
+from flash.core.model import Task
+from flash.core.registry import FlashRegistry
+from flash.core.utilities.imports import _IMAGE_STLYE_TRANSFER
+from flash.image.style_transfer import STYLE_TRANSFER_BACKBONES
+
+if _IMAGE_STLYE_TRANSFER:
+    import pystiche.demo
+    from pystiche import enc, loss, ops
+    from pystiche.image import read_image
+else:
+
+    class enc:
+        Encoder = None
+        MultiLayerEncoder = None
+
+    class ops:
+        EncodingComparisonOperator = None
+        FeatureReconstructionOperator = None
+        MultiLayerEncodingOperator = None
+
+    class loss:
+
+        class PerceptualLoss:
+            pass
+
+
+from flash.image.style_transfer.utils import raise_not_supported
+
+__all__ = ["StyleTransfer"]
+
+
+class StyleTransfer(Task):
+    """Task that transfer the style from an image onto another.
+
+    Example::
+
+        from flash.image.style_transfer import StyleTransfer
+
+        model = StyleTransfer(image_style)
+
+    Args:
+        style_image: Image or path to an image to derive the style from.
+        model: The model by the style transfer task.
+        backbone: A string or model to use to compute the style loss from.
+        content_layer: Which layer from the backbone to extract the content loss from.
+        content_weight: The weight associated with the content loss. A lower value will lose content over style.
+        style_layers: Layers from the backbone to derive the style loss from.
+        optimizer: Optimizer to use for training the model.
+        optimizer_kwargs: Optimizer keywords arguments.
+        scheduler: Scheduler to use for training the model.
+        scheduler_kwargs: Scheduler keywords arguments.
+        learning_rate: Learning rate to use for training, defaults to ``1e-3``.
+        serializer: The :class:`~flash.core.data.process.Serializer` to use when serializing prediction outputs.
+    """
+
+    backbones: FlashRegistry = STYLE_TRANSFER_BACKBONES
+
+    def __init__(
+        self,
+        style_image: Optional[Union[str, torch.Tensor]] = None,
+        model: Optional[nn.Module] = None,
+        backbone: str = "vgg16",
+        content_layer: str = "relu2_2",
+        content_weight: float = 1e5,
+        style_layers: Union[Sequence[str], str] = ("relu1_2", "relu2_2", "relu3_3", "relu4_3"),
+        style_weight: float = 1e10,
+        optimizer: Union[Type[torch.optim.Optimizer], torch.optim.Optimizer] = torch.optim.Adam,
```

Comment:
in same tasks we are promoting `AdamW` since it has a better implementation of the original algorithm. Have you ever tried here ?

Reply:
Nope. Until @tchaton changes this followed the official PyTorch example. Since we are deviating now, we could also use another optimizer.